File: frontend/src/microservices/flight/objects/FlightPlannerResponse.ts
// 本文件定义 flight 模块的 `FlightPlannerResponse`，作为 planner 响应数据并提供 JSON 编解码。

import type { FlightBookingWindowStatus } from "./FlightBookingWindowStatus"
import type { CabinInventoryResponse } from "./CabinInventoryResponse"

export type FlightPlannerResponse = {
  flightId: string
  airlineId: string
  airlineName: string
  airlineCode: string
  airlineLogoPath: string | null
  flightNumber: string
  aircraftModel: string
  departureAirport: string
  arrivalAirport: string
  departureTime: string
  arrivalTime: string
  status: string
  bookingWindowStatus: FlightBookingWindowStatus
  canBookOnline: boolean
  bookingNotice: string | null
  lateBookingSurchargeAmount: string | null
  lateBookingSurchargeCurrency: string | null
  basePrice: string
  currency: string
  createdAt: string
  cabinInventories: CabinInventoryResponse[]
}

export const flightPlannerResponseFromJson = (json: string): FlightPlannerResponse =>
  JSON.parse(json) as FlightPlannerResponse

export const flightPlannerResponseToJson = (value: FlightPlannerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/flight/objects/FlightPlannerResponse.scala
// FlightPlannerResponse defines the flight detail planner response model.
package com.typesafe.travel.flight.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class FlightPlannerResponse(
    flightId: String,
    airlineId: String,
    airlineName: String,
    airlineCode: String,
    airlineLogoPath: Option[String],
    flightNumber: String,
    aircraftModel: String,
    departureAirport: String,
    arrivalAirport: String,
    departureTime: String,
    arrivalTime: String,
    status: String,
    bookingWindowStatus: FlightBookingWindowStatus,
    canBookOnline: Boolean,
    bookingNotice: Option[String],
    lateBookingSurchargeAmount: Option[String],
    lateBookingSurchargeCurrency: Option[String],
    basePrice: String,
    currency: String,
    createdAt: String,
    cabinInventories: List[CabinInventoryPlannerResponse]
)

object FlightPlannerResponse:
  given sourceEncoder: Encoder[FlightPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[FlightPlannerResponse] = deriveDecoder

